src/App.jsx

```diff
@@ -14,33 +18,75 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const [selectedGoods, setSelectedGoods] = useState('');
+  const [reversed, setReversed] = useState(false);
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  const visibleGoods = [...goodsFromServer];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  visibleGoods.sort((good1, good2) => {
+    switch (selectedGoods) {
+      case 'button is-info':
+        return good1.localeCompare(good2);
+      case 'button is-success':
+        return good1.length - good2.length;
+      default:
+        return 0;
```

Comment:
The sorting logic is based on the button class names ('button is-info' and 'button is-success'), which is not a reliable method. It would be better to use more descriptive state values for `selectedGoods`, such as 'alphabetical' and 'length', to determine the sorting criteria.